examples/gno.land/p/demo/avlhelpers/avlhelpers.gno

```diff
@@ -0,0 +1,33 @@
+package avlhelpers
+
+import (
+	"gno.land/p/demo/avl"
+)
+
+// Get a list of keys starting from the given prefix. Limit the
+// number of results to maxResults.
+// The prefix and keys are treated as byte strings, ignoring possible multi-byte Unicode runes.
+func ListByteStringKeysByPrefix(tree avl.Tree, prefix string, maxResults int) []string {
```

Comment:
```suggestion
func ListByteStringKeysByPrefix(tree avl.Tree, prefix string, func(key string, value interface{}) bool) {
```

what about exposing this first, and eventually create another helper that does your current logic?

Reply:
I'm not sure what you mean. What does the function do? And, you removed the `[]string` return value. Was that intentional?

Reply:
Are you proposing to pause the merging of this PR to implement the new arguments?

Reply:
Hi @moul . You mean similar to the [Iterate method](https://github.com/gnolang/gno/blob/8bd07c971e7ed9ee94af03d7145f247224683694/examples/gno.land/p/demo/avl/tree.gno#L64)?

Reply:
I factored out a function `IterateByteStringKeysByPrefix` with the signature you suggest. Is this what you have in mind? https://github.com/gnolang/gno/pull/1708/commits/7f8f389ae779f68de2068e214d60557a42c6d255

Reply:
In the call, looks good.